src/main/java/com/klarna/hiverunner/HiveRunnerExtension.java

```diff
@@ -48,7 +49,7 @@
   private final HiveRunnerConfig config = new HiveRunnerConfig();
   private Path basedir;
   private HiveShellContainer container;
-  private List<? extends Script> scriptsUnderTest;
+  protected List<Script> scriptsUnderTest = new ArrayList<Script>();
```

Comment:
```suggestion
  protected List<Script> scriptsUnderTest;
```
If we didn't pre-initialise it before I don't think we need to now? Or does something not work downstream if you don't do this?

Reply:
why not introduce a getter and just keep this private?

Reply:
It doesn't work downstream if I don't initialise it, I get an error when I try to call `scriptsUnderTest.clear() `or `scriptsUnderTest.add()`

Reply:
What kind of error? A null pointer?

@patduin I guess the reasoning for not having a getter is that the contents of the list are being manipulated in the child class as opposed to just retrieved?

Reply:
ok fair enough not a big deal

Reply:
@massdosage yes, a null pointer error, although I just realized that I can leave this as `protected List<Script> scriptsUnderTest;` and initialise the variable downstream in MS; inside [this method](https://github.com/shermosa/mutant-swarm/blob/fa9d085badd810acf159d7a8b3cfdc1c6811951f/src/main/java/com/hotels/mutantswarm/MutantSwarmExtension.java#L172). Is that better?

Reply:
I guess it depends a bit on what the "contract" is here between the classes. Usually one would initialise something like this in a constructor or a method they both have via a parent/child override etc. I'm a bit concerned about what happens here, won't this call https://github.com/shermosa/mutant-swarm/blob/fa9d085badd810acf159d7a8b3cfdc1c6811951f/src/main/java/com/hotels/mutantswarm/MutantSwarmExtension.java#L131 effectively reset the list completely so in MS you'll now have lost all the changes you just made in the method above?

Reply:
Hmm, although the code is a bit weird, we pass the scripts in to the container at https://github.com/klarna/HiveRunner/blob/0d84bbf218058ceca7db56fac9a9fff328049b09/src/main/java/com/klarna/hiverunner/HiveRunnerExtension.java#L63 and then read them back from the container again at https://github.com/klarna/HiveRunner/blob/0d84bbf218058ceca7db56fac9a9fff328049b09/src/main/java/com/klarna/hiverunner/HiveRunnerExtension.java#L67. So I guess this is why they're not overridden as what you've set effectively comes back out again via the container. Anyway... I guess this all comes down to whether we want downstream classes to initialise the List or assume it's been initialised and just add/clear/remove values etc. Generally I think having an empty list is probably better as it reduces the chance of NPE's so if this all works fine in HiveRunner with this change let's keep it like this.